crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -926,26 +916,10 @@ impl CheckpointBuilder {
     async fn run(mut self) {
         info!("Starting CheckpointBuilder");
         loop {
-            // Check whether an exit signal has been received, if so we break the loop.
-            // This gives us a chance to exit, in case checkpoint making keeps failing.
-            match self.exit.has_changed() {
-                Ok(true) | Err(_) => {
-                    break;
-                }
-                Ok(false) => (),
-            };
-
             self.maybe_build_checkpoints().await;
 
-            match select(self.exit.changed().boxed(), self.notify.notified().boxed()).await {
-                Either::Left(_) => {
-                    // break loop on exit signal
-                    break;
-                }
-                Either::Right(_) => {}
-            }
+            let _ = timeout(Duration::from_secs(1), self.notify.notified()).await;
```

Comment:
I can revert to without timeout too, but it seems nice to add a bit of defense here.

Reply:
was there a timeout previously? It doesn't seem necessary imo

Reply:
There wasn't a timeout before. I remember there were issues with missing notification to checkpoint builder, so this may be useful? But it does hide potential issues and I can remove the timeout for simtests.